test/oauth2_unittest.cc

```diff
@@ -118,6 +122,38 @@ TEST_F(OAuth2Test, GetAuthHeaderValueUsingTokenFromMetadataServerAsFallback) {
   EXPECT_EQ("Bearer the-access-token", auth.GetAuthHeaderValue());
 }
 
+TEST_F(OAuth2Test, PropagateGceApiRequestErrorsCumulativeToView) {
+  testing::FakeServer oauth_server;
+  testing::TemporaryFile credentials_file(
+    std::string(test_info_->name()) + "_creds.json",
+    "{\"client_email\":\"user@example.com\",\"private_key\":\"some_key\"}");
+  Configuration config(std::istringstream(
+      "CredentialsFile: '" + credentials_file.FullPath().native() + "'\n"
+  ));
+  Environment environment(config);
+  OAuth2 auth(environment);
+  SetTokenEndpointForTest(&auth, oauth_server.GetUrl() + "/oauth2/v3/token");
+
+  // Remove existing view on kGceApiRequestErrors to avoid
+  // other tests affecting the result.
+  // Flush to propagate existing records to views, including records from other
+  // tests.
+  ::opencensus::stats::StatsExporter::RemoveView(
+      ::google::Metrics::kGceApiRequestErrors);
+  ::opencensus::stats::testing::TestUtils::Flush();
+
+  auth.GetAuthHeaderValue();
+
+  // Flush records to propagate to views.
+  ::opencensus::stats::View errors_view(
+      ::google::Metrics::GceApiRequestErrorsCumulativeViewDescriptor());
+  ::opencensus::stats::testing::TestUtils::Flush();
+
+  EXPECT_THAT(errors_view.GetData().int_data(),
```

Comment:
Can we use `::opencensus::stats::StatsExporter::GetViewData()` here and avoid the need to have a separate view?

Reply:
yes, and I found that I still need to register the ViewDescriptor for export so we can use `::opencensus::stats::StatsExporter::GetViewData()`, because the way we use singleton pattern on view registration inside `RecordGceApiRequestErrors()`. I write up the comments for my thoughts here.